src/metabase/driver/sql_jdbc/execute.clj

```diff
@@ -232,19 +232,30 @@
             (finally
               (throw e))))))))
 
+(defn- prepare-result
+  "Convert a result set to a hash-map with :rows and :columns."
+  [[columns & rows]] {:rows (or rows []) :columns (map u/qualified-name columns)})
```

Comment:
What about adding a comment to make it clear that this function preserves the laziness of the rows?

Reply:
@lins05 With the way Clojure handles laziness, I'm not sure that's necessary. I think it's more important to make it clear when lazy seqs are _realized_, rather than where they're preserved. Also, in the case where `:rows` is `[]`, such a comment could be misleading.

Reply:
Personally I'm on team @walterl here. I was surprised to find that it wasn't already lazy.